tower/src/util/either.rs

```diff
@@ -17,58 +16,73 @@ use tower_service::Service;
 /// Both services must be of the same request, response, and error types.
```

Comment:
lol, this comment was previously incorrect, but it's correct now :P

Reply:
Its was eventually consistent 🤓